src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -1344,7 +1344,15 @@ void CodeGen::genSimpleReturn(GenTree* treeNode)
         }
         else
         {
-            GetEmitter()->emitIns_R_R_I(attr == EA_4BYTE ? INS_addiw : INS_addi, attr, retReg, op1->GetRegNum(), 0);
+            if (attr == EA_4BYTE)
+            {
+                GetEmitter()->emitIns_R_R_I(INS_addiw, attr, retReg, op1->GetRegNum(), 0);
+            }
+            else
+            {
+                GetEmitter()->emitIns_R_R(INS_mov, attr, retReg, op1->GetRegNum());
+            }
```

Comment:
```suggestion
            GetEmitter()->emitIns_R_R_I(attr == EA_4BYTE ? INS_sext_w : INS_mov, attr, retReg, op1->GetRegNum());
```
nit, might as well use pseudos on both sides